Enable Canyon in the devnet

trianglesphere opened the pull request:
**Description**

This does several things to enable Canyon
1. Update to op-geth which has all canyon changes enabled in it
2. Thread the new 1559 config change through the devnet config
3. Fix some EIP-1559 calculations where were use CalcBaseFee
4. Add the PostCanyonDenominator in several deploy configs

tynes commented on packages/contracts-bedrock/deploy-config/devnetL1-template.json:
```diff
@@ -40,10 +40,12 @@
   "governanceTokenName": "Optimism",
   "governanceTokenOwner": "0xa0Ee7A142d267C1f36714E4a8F75612F20a79720",
   "eip1559Denominator": 50,
+  "eip1559DenominatorPostCanyon": 250,
```
`PostCanyon` vs just `Canyon` in the name?

trianglesphere replied:
Yea, I've been thinking about changing it. Right now it's post canyon. Canyon could be good. I was also thinking about calling it `eip1559DenominatorV2`

tynes commented on op-chain-ops/genesis/genesis.go:
```diff
@@ -31,6 +31,10 @@ func NewL2Genesis(config *DeployConfig, block *types.Block) (*core.Genesis, erro
 	if eip1559Denom == 0 {
 		eip1559Denom = 50
 	}
+	eip1559Denom2 := config.EIP1559DenominatorPostCanyon
```
`eip1559DenomCanyon` as a variable name?

ajsutton commented on packages/contracts-bedrock/deploy-config/devnetL1-template.json:
```diff
@@ -32,18 +32,20 @@
   "finalizationPeriodSeconds": 2,
   "fundDevAccounts": true,
-  "l2GenesisBlockBaseFeePerGas": "0x1",
+  "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
```
That's a very significant increase - it will take a fair while for that to drop back down to the minimum. It appears to be what's causing the test failures. :)

trianglesphere replied:
Was intended to be able to test EIP1559 params in canyon. Dropping for op-e2e b/c op-e2e is unable to handle int.

trianglesphere committed:
Enable Canyon in the devnet

This does several things to enable Canyon
1. Update to op-geth which has all canyon changes enabled in it
2. Thread the new 1559 config change through the devnet config
3. Fix some EIP-1559 calculations where were use CalcBaseFee
4. Add the PostCanyonDenominator in several deploy configs

ajsutton commented on packages/contracts-bedrock/deploy-config/devnetL1-template.json:
```diff
@@ -32,18 +32,20 @@
   "finalizationPeriodSeconds": 2,
   "fundDevAccounts": true,
-  "l2GenesisBlockBaseFeePerGas": "0x1",
+  "l2GenesisBlockBaseFeePerGas": "1",
```
Needs to be `0x1`
```suggestion
  "l2GenesisBlockBaseFeePerGas": "0x1",
```

trianglesphere committed:
Update packages/contracts-bedrock/deploy-config/devnetL1-template.json

Co-authored-by: Adrian Sutton <adrian@oplabs.co>

tynes commented on op-chain-ops/genesis/config.go:
```diff
@@ -318,6 +320,9 @@ func (d *DeployConfig) Check() error {
 	if d.EIP1559Denominator == 0 {
 		return fmt.Errorf("%w: EIP1559Denominator cannot be 0", ErrInvalidDeployConfig)
 	}
+	if d.L2GenesisCanyonTimeOffset != nil && d.EIP1559DenominatorCanyon == 0 {
```
We could move to a model where the eip1559 params are optional and it logs it as an override if present, otherwise it goes with the default value

trianglesphere replied:
making a note of this.
